vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -265,8 +277,19 @@ def __init__(
 
         self.tp_size = (tp_size if tp_size is not None else
                         get_tensor_model_parallel_world_size())
+        self.ep_size = (ep_size if ep_size is not None else
+                        get_expert_model_parallel_size())
+        if num_experts % self.ep_size != 0:
+            raise ValueError(
+                f"Number of routed experts {num_experts} must be "
+                f"divisible by the expert parallel size {self.ep_size}.")
+        if self.tp_size % self.ep_size != 0:
+            raise ValueError(
+                f"Tensor parallel size {self.tp_size} must be divisible by "
+                f"the expert parallel size {self.ep_size}.")
```

Comment:
I feel this message may be confusing, because the TP size here is not the actual TP size provided by `--tp`. IIUC, it's `tp*ep`. Thus this to me is more like an internal error, so we should just use assert.